csrc/python_frontend/fusion_definition.cpp

```diff
@@ -351,13 +351,18 @@ std::optional<size_t> FusionDefinition::id() const {
 
 Scalar FusionDefinition::defineScalar() {
   FUSER_PERF_SCOPE("FusionDefinition::defineScalar");
+  NVF_CHECK(
+      trie_node_ != nullptr,
+      "defineScalar must be called inside a 'with' block");
```

Comment:
```suggestion
      "define_scalar() must be called from an initialized definition via a python context manager or a child class' definition() method.");
```

Reply:
Note, I am purposely suggesting using the python function name `define_scalar()` to give a better error message given this error is most likely to be seen from Python.